packages/account/src/providers/provider.ts

```diff
@@ -1150,6 +1152,8 @@ Supported fuel-core version: ${supportedVersion}.`
     const { transactionRequest } = params;
     let { gasPrice } = params;
 
+    await this.fetchChainAndNodeInfo(true);
```

Comment:
Would it be easier to keep everything as it was and have this here instead?

```suggestion
    await Provider.clearChainAndNodeCaches();
    await this.fetchChainAndNodeInfo();
```

Reply:
On a second look, invalidating the cache on every tx estimation doesn't seem as efficient as only when the consensus version changes.

Reply:
You're right, and it's definitely introduced some performance regressions, better to monitor the consensus version.

Reply:
On a third look, this blocks every tx estimation and degrades the experience. This check must be non-blocking.


```ts
// no await / non-blocking
Provider.autoRefreshCache().catch(..);
```

Then:

```ts
// pseudo-code
function autoRefreshCache () {
  const previous = Provider.consensusParamsVersion;
  const current = Provider.getConsensusVersion();

  if (previous !== current) {
    Provider.clearChainAndNodeCaches();
    await Provider.fetchChainAndNodeInfo();
  }
}
```

Reply:
Btw, this seems more like a feature/enhancement than a bug/fix.

Reply:
Good idea @arboleya c3eadcea4